src/ImageSharp/Formats/Webp/Lossy/YuvConversion.cs

```diff
@@ -312,6 +586,178 @@ public static void YuvToBgr(int y, int u, int v, Span<byte> bgr)
+            Vector128<byte> shuffle2,
+            out Vector128<byte> output0,
+            out Vector128<byte> output1,
+            out Vector128<byte> output2,
+            out Vector128<byte> output3,
+            out Vector128<byte> output4,
+            out Vector128<byte> output5)
+        {
+            output0 = Ssse3.Shuffle(input0, shuffle0);
+            output1 = Ssse3.Shuffle(input0, shuffle1);
+            output2 = Ssse3.Shuffle(input0, shuffle2);
+            output3 = Ssse3.Shuffle(input1, shuffle0);
+            output4 = Ssse3.Shuffle(input1, shuffle1);
+            output5 = Ssse3.Shuffle(input1, shuffle2);
+        }
+
+        // Convert 32 samples of YUV444 to R/G/B
+        private static void ConvertYuv444ToRgbSse41(Span<byte> y, Span<byte> u, Span<byte> v, out Vector128<short> r, out Vector128<short> g, out Vector128<short> b)
+        {
+            Vector64<byte> yTmp = Unsafe.As<byte, Vector64<byte>>(ref MemoryMarshal.GetReference(y));
+            Vector64<byte> uTmp = Unsafe.As<byte, Vector64<byte>>(ref MemoryMarshal.GetReference(u));
+            Vector64<byte> vTmp = Unsafe.As<byte, Vector64<byte>>(ref MemoryMarshal.GetReference(v));
+            Vector128<byte> y0 = LoadHigh(yTmp);
+            Vector128<byte> u0 = LoadHigh(uTmp);
+            Vector128<byte> v0 = LoadHigh(vTmp);
+
+            Vector128<ushort> y1 = Sse2.MultiplyHigh(y0.AsUInt16(), K19077.AsUInt16());
+            Vector128<ushort> r0 = Sse2.MultiplyHigh(v0.AsUInt16(), K26149.AsUInt16());
+            Vector128<ushort> g0 = Sse2.MultiplyHigh(u0.AsUInt16(), K6419.AsUInt16());
+            Vector128<ushort> g1 = Sse2.MultiplyHigh(v0.AsUInt16(), K13320.AsUInt16());
+
+            Vector128<ushort> r1 = Sse2.Subtract(y1.AsUInt16(), K14234.AsUInt16());
+            Vector128<ushort> r2 = Sse2.Add(r1, r0);
+
+            Vector128<ushort> g2 = Sse2.Add(y1.AsUInt16(), K8708.AsUInt16());
+            Vector128<ushort> g3 = Sse2.Add(g0, g1);
+            Vector128<ushort> g4 = Sse2.Subtract(g2, g3);
+
+            Vector128<ushort> b0 = Sse2.MultiplyHigh(u0.AsUInt16(), K33050.AsUInt16());
+            Vector128<ushort> b1 = Sse2.AddSaturate(b0, y1);
+            Vector128<ushort> b2 = Sse2.SubtractSaturate(b1, K17685.AsUInt16());
+
+            // Use logical shift for B2, which can be larger than 32767.
+            r = Sse2.ShiftRightArithmetic(r2.AsInt16(), 6); // range: [-14234, 30815]
+            g = Sse2.ShiftRightArithmetic(g4.AsInt16(), 6); // range: [-10953, 27710]
+            b = Sse2.ShiftRightLogical(b2.AsInt16(), 6); // range: [0, 34238]
+        }
+
+        // Load the bytes into the *upper* part of 16b words. That's "<< 8", basically.
+        [MethodImpl(InliningOptions.ShortMethod)]
+        private static Vector128<byte> LoadHigh(Vector64<byte> src)
+        {
+            Vector128<byte> tmp = Unsafe.As<Vector64<byte>, Vector128<byte>>(ref src);
+            return Sse2.UnpackLow(Vector128<byte>.Zero, tmp);
```

Comment:
Can't you actually load to `Vector128<byte>`, then bitshift? Could be much cheaper, not sure what is `Unsafe.As<Vector64<byte>, Vector128<byte>>(ref src)` being compiled to.

Reply:
Its just a `mov`, [SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCEBXAMy5ilwDcNek1YAlDgDsMAS3wwWASWlQZk3DLCDhjZiwnS5C5RlXrNuFgA0AHEiG1d4qbPksAwhHwAHGQBs+AGU+ADdNGG1aAGYGKVxsXiZSBncGAG8aBiymGOIUBgBZAAoANRgwDGhUAB5gAE8MGAA+BlwoMABKBkzsjOpsgYYyiuhyUhtahuaGDB8GAF4GAFV1BIUAQVxq4cqoGvrGprQh8t2xiYPmpqLYLlb2jocBgF8aZ6A=), this should be fine, right?

Reply:
I cant seem to figure out how to do this differently/better.

Reply:
I thought this is a vector level shuffle/shift, now I see what you are doing. What I mean is that if you only unpack from the lower half of `tmp`, pushing this through `Vector64<byte>` can be avoided by adding an 8 byte padding to `y`, `u`, `v`:

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCEBXAMy5ilwDcNek1YAlDgDsMAS3wwWASWlQZk3DLCDhjZiwnS5C5RlXrNuFgA0AHEiG1d4qbPksAwhHwAHGQBs+AGU+ADdNGG1aAGYGKVxsXiZSBncGAG8aBiymGIA1GDAMaHJSGwAeYABPDBgAPgYAQQAKWC4GKpqGXCgwAEoM6myhhnzC6FQK6rqGDB9yBgBeBgBVdQSFBtxJmrQRgqKoCY662paYNu6+h2Hs0YOS8uP62e9kpdX43hZNsrvxlG2dV2fygD0BtVOrRmc161xuTAA7AxArgYKQWKtvNgwABrAAyEAA7k0QWCniwAFp8CC7F6kWGZbIAX0ZWVZOT2Y1BpXBDDYZzaxy6PX67KGpJ5bCmzx8ixWay+P2OwP2xR5T0h52FVzF2WISJRaIxkixuIJxIlj2llOptJ8DMGzJoTKAA===

Reply:
I was not aware of this, looks much better now, thanks.

Changed with cc5f7af7